integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/statistics/ProxyStatisticsTest.java

```diff
@@ -71,6 +71,39 @@ public void statisticsCollectionCountTest() throws Exception {
 
     }
 
+    @Test(groups = {"wso2.esb"}, description = "Dynamic endpoint stat check")
+    public void dynamicEndpointStatisticsCollectionTest() throws Exception {
+
+        thriftServer.resetMsgCount();
+        thriftServer.resetPreservedEventList();
+        axis2Client.sendSimpleStockQuoteRequest(getProxyServiceURLHttp("TestEndpointStatProxy"), null, "WSO2");
+        thriftServer.waitToReceiveEvents(2000, 1);
+        Assert.assertEquals(thriftServer.getMsgCount(), 1,
+                "Hundred statistics events are required, but different number " +
+                        "is found");
+        Map<String, Object> aggregatedEvent =
+                ESBTestCaseUtils.decompress((String) thriftServer.getPreservedEventList().get(0).getPayloadData()[1]);
+        ArrayList eventList = (ArrayList) aggregatedEvent.get("events");
+        HashSet<String> allMediatorEventIds = new HashSet<>();
+        for (Object list : eventList) {
+            allMediatorEventIds.add((String) ((ArrayList) list).get(MEDIATOR_ID_INDEX));
+        }
+
+        ArrayList<String> mediatorList = new ArrayList<>();
+        mediatorList.add("TestEndpointStatProxy@0:TestEndpointStatProxy");
+        mediatorList.add("TestEndpointStatProxy@1:PROXY_INSEQ");
+        mediatorList.add("TestEndpointStatProxy@2:CallMediator");
+        mediatorList.add("TestEndpointStatProxy@3:StatEndpoint1");
+        mediatorList.add("TestEndpointStatProxy@4:RespondMediator");
+
+        //Checking whether all the mediators are present in the event
+        Assert.assertEquals(eventList.size(), 5, "Four configuration events are required");
```

Comment:
four or five ?

Reply:
Updated the message with [5191446](https://github.com/wso2/micro-integrator/pull/2700/commits/519144676b3ab210de454c6e8570859c21cb535e)